source/feature/openvswitch/openvswitch-spec.html.md

```diff
@@ -0,0 +1,183 @@
+1. Replace native linux networking entities (e.g. bridge) with OVS on oVirt hosts.
+2. Preserve current networking requirements from oVirt hosts.
+3. Preserve or increase networking performance.
+4. Keep both switching options available, letting end users of ovirt-4.0 decide when they would like to upgrade their existing 3.6 clusters.
+
+## Overview
+The introduction of OVS as an additional network switching solution raises the need to generalize and abstract the requirements from the system networking services and define a spec which considers both options (ovs & native linux networking).
+The OVS model of arranging networking entities is different from the native linux networking, but is closer to industry networking modeling.
+
+### Network Entities
+In the following table, networking entities are listed with their implementation options.
+
+|     Entity       | Kernel |  OVS  |    Remark     |
+| :----------------| :----: | :---: | ------------- |
+| Interface        |   X    |   X*  | Interfaces may be created by OVS on the kernel and attached to OVS ports. |
+| Port/NIC         |   X    |   X   |               |
+| Bond             |   X    |   X   | kernel mod    |
+| Bridge           |   X    |   X   |               |
+| Tagging & Tunneling | X   |   X   |               |
+| QoS              |   X    |   X   |               |
+| IP routing       |   X    | w/Controller |        |
+| TCP/UDP stack    |   X    |   -   |               |
+
+### Linux native networking model
+![Linux native networking model](linux_native_networking_model.jpg)
+
+### OVS networking model 
+![OVS networking model](ovs_networking_model.jpg)
+
+
+## Specifications
+
+### Solution Diagram
+![Solution Diagram](solution_diagram.jpg)
+
+### NetSwitch
+A generic switch interface, relaying setup or report requests to the relevant switch implementation (currently two, legacy linux networking and ovs).
+
+This aproach is prefered over having an additional configurator type under the existing switch scheme:
+
+- Adding OVS as a configurator may require changes in the upper common levels, which may introduce instability in existing functionality.
+The proposed abstraction level is aimed to leave existing logic isolated from OVS (and vice versa), reducing potential regression and instability.
+- Different switches or domains should be decoupled, each having their own underling implementation options allowing full flexability on how they are implemented.
+- The modeling of each switch is different, having it seperated reduces complex logic.
+
+Switch implementations are registered per availability, with an identification and a callback. 
+setupNetworks requests are expected to arrive from Engine with a switch type ID, based on which the request is forwarded to the switch implementation.
+
+
+### OVS Configurators
+There are at least two ways to configure OVS: Using the command line utilities (i.e ovs-vsctl) and OVSDB protocol python binding. The latter is considered faster.
```

Comment:
I believe that this is what @mykaul referred to: the textual output of ovs-vsctl may supposedly change without prior notice; it is not considered as a stable API by its developers.